app/controllers/api_controller/parser/request_adapter.rb

```diff
@@ -39,15 +41,16 @@ def c_suffix
       end
 
       def c_id
-        @params[:c_id] || c_path_parts[1]
+        id = @params[:c_id] || c_path_parts[1]
```

Comment:
can't we do from_cid(@params[:c_id] || c_path_parts[1]) here like we do for s_id ?